docs/cli/dnx-migration.md

```diff
@@ -0,0 +1,152 @@
+in-memory compilation. It will actually invoke `dotnet build` to build your code and then run the built binary. 
+
+Another way is using the `dotnet` itself to run your code. This is done by providing a path to your assembly:
+`dotnet path/to/an/assembly.dll`. 
+
+## Migrating your DNX project to .NET Core CLI
+In addition to using new commands when working with your code, there are two major thing left in migrating from DNX:
+
+1. Migrating the project file (`project.json`) itself to the CLI tooling.
+2. Migrating off of any DNX APIs to their BCL counterparts. 
+
+### Migrating the project file
+The CLI and DNX both use the same basic project system based on `project.json` file. The syntax and the semantics of the 
+project file are pretty much the same, with mild differences based on the scenarios. Let's see what those differences are. 
+
+If you are building a console application, you need to add the following snippet to your project file:
+
+```json
+"compilationOptions": {
+    "emitEntryPoint": true
+}
+```
+
+This instructs `dotnet build` to emit an entry point for your application, effectively making your code runnable. If 
+you are building a class library, simply omit the above section. Of course, once you add the above snippet to your 
+`project.json` file, you need to add a static entry point; with the move off DNX, the DI services it provided are no 
+longer available and thus this needs to be a basic .NET entry point: `static void Main()`.
+
+If you have a "commands" section in your `project.json`, you can remove it. Some of the commands that used to exist as 
+DNU commands, such as Entity Framework CLI commands, are being ported to be 
+per-project extensions to the CLI. If you built your own commands that you are using in your projects, you need to 
+replace them with CLI extensions. In this case, the `commands` node in `project.json` needs to be replaced by the 
+`tools` node and it needs to list the tools dependencies as explained in the 
+[CLI extensibility section](overview.md#extensibility). 
+
+After these things are done, you need to decide which type of portability you wish for you app. With .NET Core, we have 
+invested into providing a spectrum of portability options that you can choose from. For instance, you may want to have 
+a fully *portable* application or you may want to have a *self-contained* application. The former is more akin to the way 
+you've most likely seen .NET application to function: it needs a shared component to execute it on the target machine 
+(the .NET Core runtime). The latter doesn't require that, but you have to produce one application for each OS you wish to support. 
+These portability types and more are discussed in the [application portability type](app-types.md) document. 
+
+Once you make a call on what type of portability you want, you need to change your targeted framework(s). If you were 
```

Comment:
"what type of portability you want" - I'd reword this to "how portable you want your application to be"